### PR TITLE
Port get_notifications RPC to RPC marshaller

### PR DESCRIPTION
**Pull request by Quexington**
This moves a single RPC to the new RPC `@marshal` decorator.

### CODE REVIEW

File: chia/rpc/wallet_rpc_api.py [hunk 1]
```diff
@@ -1407,34 +1408,22 @@ async def extend_derivation_index(self, request: Dict[str, Any]) -> Dict[str, An
...
-                {
-                    "id": notification.coin_id.hex(),
-                    "message": notification.message.hex(),
-                    "amount": notification.amount,
-                    "height": notification.height,
-                }
-                for notification in notifications
-            ]
-        }
+        return GetNotificationsResponse(notifications)
```
Comment (altendky) [inline chia/rpc/wallet_rpc_api.py:1426]: This is changing the `id` key to `coin_id`.

File: chia/rpc/wallet_rpc_client.py [hunk 2]
```diff
@@ -1258,27 +1258,8 @@ async def dl_delete_mirror(
...
-            Notification(
-                bytes32.from_hexstr(notification["id"]),
-                bytes.fromhex(notification["message"]),
-                uint64(notification["amount"]),
-                uint32(notification["height"]),
-            )
-            for notification in response["notifications"]
-        ]
+    async def get_notifications(self, request: GetNotifications) -> GetNotificationsResponse:
+        return GetNotificationsResponse.from_json_dict(await self.fetch("get_notifications", request.to_json_dict()))
```
Comment (altendky) [inline chia/rpc/wallet_rpc_client.py:1262]: This is changing the `id` key to `coin_id`.

**Review by altendky — CHANGES_REQUESTED** [review]

File: chia/rpc/wallet_rpc_api.py (hunk 1 above)
Comment (Quexington) [reply chia/rpc/wallet_rpc_api.py:1426]: fixed by renaming streamable member

File: chia/rpc/wallet_rpc_client.py (hunk 2 above)
Comment (Quexington) [reply chia/rpc/wallet_rpc_client.py:1262]: fixed by renaming streamable member

File: chia/rpc/wallet_request_types.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+from chia.util.streamable import Streamable, streamable
+from chia.wallet.notification_store import Notification
+
+
+@streamable
+@dataclass(frozen=True)
+class GetNotifications(Streamable):
+    ids: Optional[List[bytes32]] = None
+    start: Optional[uint32] = None
+    end: Optional[uint32] = None
```
[the remark this answers is not in the document]
Comment (Quexington) [reply chia/rpc/wallet_request_types.py:17]: They do yeah, Streamable's from_json_dict takes care of that

File: chia/cmds/wallet_funcs.py [hunk 4]
```diff
@@ -1382,10 +1383,14 @@ async def get_notifications(
         if ids is not None and len(ids) == 0:
             ids = None
 
-        notifications = await wallet_client.get_notifications(ids=ids, pagination=(start, end))
-        for notification in notifications:
+        response = await wallet_client.get_notifications(
+            GetNotifications(
+                ids=ids, start=None if start is None else uint32(start), end=None if end is None else uint32(end)
```
Comment (altendky) [inline chia/cmds/wallet_funcs.py:1388]: This is new.

```suggestion
                ids=ids, start=uint32.construct_optional(start), end=uint32.construct_optional(end)
```

File: chia/rpc/wallet_rpc_api.py [hunk 5]
```diff
@@ -1407,34 +1408,22 @@ async def extend_derivation_index(self, request: Dict[str, Any]) -> Dict[str, An
...
         return {"success": True, "index": updated_index}
 
-    async def get_notifications(self, request: Dict[str, Any]) -> EndpointResult:
-        ids: Optional[List[str]] = request.get("ids", None)
-        start: Optional[int] = request.get("start", None)
-        end: Optional[int] = request.get("end", None)
-        if ids is None:
+    @marshal
+    async def get_notifications(self, request: GetNotifications) -> GetNotificationsResponse:
+        if request.ids is None:
```
Comment (altendky) [inline chia/rpc/wallet_rpc_api.py:1413]: This is not a thought to be addressed now.  Feel free to close this conversation after reading it.

Silently ignoring either `ids` or `start` and `end` when the other is specified is bound to cause someone some debugging hassle.  More complete schema definition etc that would make them exclusive, or actually handling both when both are requested, would be nice.  Mostly this is likely a general though.  Maybe even separate endpoints would be more appropriate.

**Review by altendky — CHANGES_REQUESTED** [review]

**Review by altendky — DISMISSED** [review]

**Review by altendky — APPROVED** [review]

**Review by wjblanke — APPROVED** [review]
aok

**Pull request merged**